### PR TITLE
Richie hamburger mobile UI

### PR DESCRIPTION
**Pull request by richardjhong**
- adding hamburger-react package
- creating separate MobileHeader and DesktopHeader folders within parent Header folder
- adding media query breakpoints (targeting sm), changing logic to not show Connect Wallet button when below threshold
- adding eventListener to change currentTab to Vaulted when on mobile

### CODE REVIEW

File: src/app/Header/MobileHeader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import Link from "next/link";
+import React, { useEffect, useState } from "react";
+import { useAccount } from "wagmi";
+import { NetworkSwitch } from "@/app/Connect";
+import { DriftLogo } from "@/app/icons/DriftLogo";
+import Twitter from "@/app/icons/Twitter";
+import { Sling as Hamburger } from "hamburger-react";
+import "./header.css";
+
+type NavItem = {
```
Comment (nickick) [inline src/app/Header/MobileHeader/index.tsx]: this should be DRY'd up, import the `NavItem` type from somewhere else or export it to the other nav headers from here. I'd guess you'd want to export the type from `src/app/Header/index.tsx`.

File: src/app/Vault/index.tsx [hunk 2]
```diff
@@ -52,6 +52,25 @@ const tabExplanations: { [key in TabNames]: ReactNode } = {
...
+
+    window.addEventListener("resize", handleResize);
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (windowWidth <= 640 && currentTab === TabNames.VAULTED) {
```
Comment (nickick) [inline src/app/Vault/index.tsx]: abstract out the magic number into it's own variable in caps to make this easy to understand / configure.

so something like

```
const MOBILE_WIDTH = 640;

useEffect(...
```

Comment (nickick) [reply src/app/Vault/index.tsx]: Also, you could consider using a `useIsMobile` hook to abstract this out and make this more declarative

File: package-lock.json
Comment (nickick) [inline package-lock.json:1]: Make sure you're using yarn, not npm, to install things. Delete this file.

File: yarn.lock [hunk 3]
```diff
@@ -3656,7 +3661,7 @@ lodash.merge@^4.6.2:
 
 lodash@^4.17.20, lodash@^4.17.21:
   version "4.17.21"
-  resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.21.tgz#679591c564c3bffaae8454cf0b3df370c3d6911c"
+  resolved "https://registry.npmjs.org/lodash/-/lodash-4.17.21.tgz"
```
Comment (nickick) [inline yarn.lock:3664]: I think some of these changes are using npm vs yarn. Use yarn for this package.

**Review by nickick — CHANGES_REQUESTED** [review]
Mostly nits, let's resolve before pulling this in.

File: src/app/Header/MobileHeader/index.tsx (hunk 1 above)
Comment (richardjhong) [reply src/app/Header/MobileHeader/index.tsx]: exporting the original NavItem from DesktopHeader.tsx, should be DRY now

File: src/app/Vault/index.tsx (hunk 2 above)
Comment (richardjhong) [reply src/app/Vault/index.tsx]: added MOBILE_WIDTH and migrated logic into useIsMobile hook

File: package-lock.json
Comment (richardjhong) [reply package-lock.json:1]: deleted

File: yarn.lock (hunk 3 above)
Comment (richardjhong) [reply yarn.lock:3664]: deleted node_modules and installed via yarn install, should be good now

**Review by nickick — APPROVED** [review]
Looks good, pulling this in.

**Pull request merged**